src/index.ts

```diff
@@ -380,8 +380,8 @@ class ExpectedRequest {
      * @param {Boolean} rawBody true if the response should be returned directly rather
      * than json-stringifying it first.
      */
-    public respond = (
-        code: number, data?: Record<string, unknown> | (() => Record<string, unknown>),
+    public respond = <T = Record<string, unknown>>(
+        code: number, data?: T | ((...props: any[]) => T),
```

Comment:
Can we type this better based on the actual call signature - https://github.com/matrix-org/matrix-mock-request/blob/69485aa41beafde41567d8784c1f50a931da54a1/src/index.ts#L279